[BUG] (PSU_Control without Hotend)

### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

Enabled #define PSU_CONTROL
#define AUTO_POWER_CONTROL
#define POWER_TIMEOUT  

Only using #define TEMP_SENSOR_COOLER 1

Have no Extruder or Hotend because i am using Laser, so i disabled it.

When i enabled auto_power_control the code copliles with error:
`Marlin\src\feature\power.cpp: In static member function 'static bool Power::is_power_needed()':
Marlin\src\feature\power.cpp:84:77: error: 'class Temperature' has no member named 'temp_hotend'
   84 |   HOTEND_LOOP() if (thermalManager.degTargetHotend(e) > 0 || thermalManager.temp_hotend[e].soft_pwm_amount > 0) return true;
      |                                                                             ^~~~~~~~~~~`
With auto_power_control disabled it compiles fine.



### Bug Timeline

new

### Expected behavior

compile with no errors.
(After 30 seconds M81 in idle.)


### Actual behavior

compile with error

### Steps to Reproduce

#define PSU_CONTROL
#define AUTO_POWER_CONTROL

disable extruder and any hotend stuff

### Version of Marlin Firmware

latest github

### Printer model

-

### Electronics

-

### Add-ons

-

### Your Slicer

_No response_

### Host Software

_No response_

### Additional information & file uploads

I am unsure about the function PSU_CONTROL maybe i missed something in the code?
Attached my config.
[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/6721323/Marlin.zip)

--- a/Marlin/src/feature/power.cpp
+++ b/Marlin/src/feature/power.cpp
@@ -62,11 +62,13 @@ bool Power::is_power_needed() {
     if (controllerFan.state()) return true;
   #endif
 
-  if (TERN0(AUTO_POWER_CHAMBER_FAN, thermalManager.chamberfan_speed))
-    return true;
+  #if ENABLED(AUTO_POWER_CHAMBER_FAN)
+    if (thermalManager.chamberfan_speed) return true;
+  #endif
 
-  if (TERN0(AUTO_POWER_COOLER_FAN, thermalManager.coolerfan_speed))
-    return true;
+  #if ENABLED(AUTO_POWER_COOLER_FAN)
+    if (thermalManager.coolerfan_speed) return true;
+  #endif
 
   // If any of the drivers or the bed are enabled...
   if (X_ENABLE_READ() == X_ENABLE_ON || Y_ENABLE_READ() == Y_ENABLE_ON || Z_ENABLE_READ() == Z_ENABLE_ON
@@ -85,18 +87,23 @@ bool Power::is_power_needed() {
     #endif
   ) return true;
 
-  HOTEND_LOOP() if (thermalManager.degTargetHotend(e) > 0 || thermalManager.temp_hotend[e].soft_pwm_amount > 0) return true;
-  if (TERN0(HAS_HEATED_BED, thermalManager.degTargetBed() > 0 || thermalManager.temp_bed.soft_pwm_amount > 0)) return true;
+  #if HAS_HOTEND
+    HOTEND_LOOP() if (thermalManager.degTargetHotend(e) > 0 || thermalManager.temp_hotend[e].soft_pwm_amount > 0) return true;
+  #endif
 
-  #if HAS_HOTEND && AUTO_POWER_E_TEMP
+  #if HAS_HEATED_BED
+  if (thermalManager.degTargetBed() > 0 || thermalManager.temp_bed.soft_pwm_amount > 0) return true;
+  #endif
+
+  #if HAS_HOTEND && ENABLED(AUTO_POWER_E_TEMP)
     HOTEND_LOOP() if (thermalManager.degHotend(e) >= (AUTO_POWER_E_TEMP)) return true;
   #endif
 
-  #if HAS_HEATED_CHAMBER && AUTO_POWER_CHAMBER_TEMP
+  #if HAS_HEATED_CHAMBER && ENABLED(AUTO_POWER_CHAMBER_TEMP)
     if (thermalManager.degChamber() >= (AUTO_POWER_CHAMBER_TEMP)) return true;
   #endif
 
-  #if HAS_COOLER && AUTO_POWER_COOLER_TEMP
+  #if HAS_COOLER && ENABLED(AUTO_POWER_COOLER_TEMP)
     if (thermalManager.degCooler() >= (AUTO_POWER_COOLER_TEMP)) return true;
   #endif
 

--- a/Marlin/src/feature/power.cpp
+++ b/Marlin/src/feature/power.cpp
@@ -62,13 +62,11 @@ bool Power::is_power_needed() {
     if (controllerFan.state()) return true;
   #endif
 
-  #if ENABLED(AUTO_POWER_CHAMBER_FAN)
-    if (thermalManager.chamberfan_speed) return true;
-  #endif
+  if (TERN0(AUTO_POWER_CHAMBER_FAN, thermalManager.chamberfan_speed))
+    return true;
 
-  #if ENABLED(AUTO_POWER_COOLER_FAN)
-    if (thermalManager.coolerfan_speed) return true;
-  #endif
+  if (TERN0(AUTO_POWER_COOLER_FAN, thermalManager.coolerfan_speed))
+    return true;
 
   // If any of the drivers or the bed are enabled...
   if (X_ENABLE_READ() == X_ENABLE_ON || Y_ENABLE_READ() == Y_ENABLE_ON || Z_ENABLE_READ() == Z_ENABLE_ON
@@ -91,19 +89,17 @@ bool Power::is_power_needed() {
     HOTEND_LOOP() if (thermalManager.degTargetHotend(e) > 0 || thermalManager.temp_hotend[e].soft_pwm_amount > 0) return true;
   #endif
 
-  #if HAS_HEATED_BED
-  if (thermalManager.degTargetBed() > 0 || thermalManager.temp_bed.soft_pwm_amount > 0) return true;
-  #endif
+  if (TERN0(HAS_HEATED_BED, thermalManager.degTargetBed() > 0 || thermalManager.temp_bed.soft_pwm_amount > 0)) return true;
 
-  #if HAS_HOTEND && ENABLED(AUTO_POWER_E_TEMP)
+  #if HAS_HOTEND && AUTO_POWER_E_TEMP
     HOTEND_LOOP() if (thermalManager.degHotend(e) >= (AUTO_POWER_E_TEMP)) return true;
   #endif
 
-  #if HAS_HEATED_CHAMBER && ENABLED(AUTO_POWER_CHAMBER_TEMP)
+  #if HAS_HEATED_CHAMBER && AUTO_POWER_CHAMBER_TEMP
     if (thermalManager.degChamber() >= (AUTO_POWER_CHAMBER_TEMP)) return true;
   #endif
 
-  #if HAS_COOLER && ENABLED(AUTO_POWER_COOLER_TEMP)
+  #if HAS_COOLER && AUTO_POWER_COOLER_TEMP
     if (thermalManager.degCooler() >= (AUTO_POWER_COOLER_TEMP)) return true;
   #endif
 

